File: client2/src/game/components/Damage.ts
import { ComponentFlag } from '../flags'

export class Damage {
  static _id = 'Damage'
  _id = Damage._id

  static _flag = ComponentFlag.Damage
  _flag = ComponentFlag.Damage

  hitpoints: number

  constructor(hitpoints: number = 0) {
    this.hitpoints = hitpoints
  }
}

File: packages/client2/lib/game/components/damage.dart
import '../../ecs/component.dart';
import '../flags.dart';

class Damage extends Component {
  static const int Flag = ComponentFlags.Damage;
  int flag = Damage.Flag;

  static const String Id = 'Damage';
  String id = Damage.Id;

  int hitpoints;

  Damage({
    this.hitpoints,
  });

  @override
  String toString() {
    return 'Damage (hp: $hitpoints)';
  }
}
